src/core/border_router/router_advertisement.hpp

```diff
@@ -0,0 +1,382 @@
+     * @param[in]  aBuffer        The buffer within which the options are held.
+     * @param[in]  aBufferLength  The length of the buffer.
+     *
+     * @returns  The address of the next option if there are a valid one. Otherwise, nullptr.
+     *
+     */
+    static const Option *GetNextOption(const Option *aCurOption, const uint8_t *aBuffer, uint16_t aBufferLength);
+
+    /**
+     * This method tells whether this option is valid.
+     *
+     * @return  A boolean indicates whether this option is valid.
+     *
+     */
+    bool IsValid(void) const { return mLength > 0; }
+
+private:
+    Type    mType;   // Type of the option.
+    uint8_t mLength; // Length of the option in unit of 8 octets,
+                     // including the `type` and `length` fields.
+} OT_TOOL_PACKED_END;
+
+/**
+ * This class represents the Prefix Information Option.
+ *
+ * See section 4.6.2 of RFC 4861 for definition of this option.
+ * https://tools.ietf.org/html/rfc4861#section-4.6.2
+ *
+ */
+OT_TOOL_PACKED_BEGIN
+class PrefixInfoOption : public Option
+{
+public:
+    /**
+     * This constructor initializes this option with zero prefix
+     * length, valid lifetime and preferred lifetime.
+     *
+     */
+    PrefixInfoOption(void);
+
+    /**
+     * This method sets the on-link (L) flag.
+     *
+     * @param[in]  aOnLink  A boolean indicates whether the prefix is on-link or off-link.
+     *
+     */
+    void SetOnLink(bool aOnLink);
+
+    /**
+     * This method sets the autonomous address-configuration (A) flag.
+     *
+     * @param[in]  aAutoAddrConfig  A boolean indicates whether this prefix can be used
+     *                              for SLAAC.
+     *
+     */
+    void SetAutoAddrConfig(bool aAutoAddrConfig);
+
+    /**
+     * This method set the valid lifetime of the prefix in seconds.
+     *
+     * @param[in]  aValidLifetime  The valid lifetime in seconds.
+     *
+     */
+    void SetValidLifetime(uint32_t aValidLifetime) { mValidLifetime = HostSwap32(aValidLifetime); }
+
+    /**
+     * THis method returns the valid lifetime of the prefix in seconds.
+     *
+     * @return  The valid lifetime in seconds.
+     *
+     */
+    uint32_t GetValidLifetime(void) const { return HostSwap32(mValidLifetime); }
+
+    /**
+     * This method sets the preferred lifetime of the prefix in seconds.
+     *
+     * @param[in]  aPreferredLifetime  The preferred lifetime in seconds.
+     *
+     */
+    void SetPreferredLifetime(uint32_t aPreferredLifetime) { mPreferredLifetime = HostSwap32(aPreferredLifetime); }
+
+    /**
+     * This method sets the prefix.
+     *
+     * @param[in]  aPrefix  The prefix contained in this option.
+     *
+     */
+    void SetPrefix(const Ip6::Prefix &aPrefix);
+
+    /**
+     * THis method returns the prefix in this option.
+     *
+     * @param[out]  aPrefix  The prefix to output to.
+     *
+     */
+    void GetPrefix(Ip6::Prefix &aPrefix) const;
```

Comment:
We can't do this because we don't have a `Ip6::Prefix` member because has PIO stores its prefix length at the beginning. The `mPrefix` member is actually type of `Ip6::Address`.